src/lib/content/index.ts

```diff
@@ -0,0 +1,74 @@
+import { getCollection as getAstroCollection, getEntry as getAstroCollectionEntry } from 'astro:content';
+import { getLocale, isLocale } from '@lib/i18n';
+import { SiteLocale } from '@lib/datocms/types';
+import { collectionMap } from '@content/config';
+
+export type CollectionKey = keyof typeof collectionMap;
+export type CollectionEntry<Key extends CollectionKey> =
+  Awaited<ReturnType<typeof getAstroCollection<Key>>>[number];
+
+/**
+ * Fetches entries from a collection.
+ *
+ * @param collection - The key of the collection to fetch entries from
+ * @param filter - Optional SiteLocale or function to filter the collection entries
+ * @returns A promise that resolves to an array of normalized collection entries
+ */
+export async function getCollection<Key extends CollectionKey>(
+  collection: Key,
+  filter?: ((entry: CollectionEntry<Key>) => boolean),
+  locale: SiteLocale | null = getLocale(),
+): Promise<CollectionEntry<Key>[]> {
+  // Filter on locale and passed filter function
+  const filterFunction = (!filter && !locale)
+    ? undefined
+    : (entry: CollectionEntry<Key>) => {
+      const entryLocale = split(entry.id).locale;
+      return [
+        // Check if the entry's locale is set and matches the requested locale
+        !locale || !entryLocale || entryLocale === locale,
+        // If a filter function is provided, apply it to the entry
+        !filter || (typeof filter === 'function' && filter(entry)),
+      ].every(Boolean);
+    };
+
+  return getAstroCollection(collection, filterFunction);
+}
+
+/**
+ * Fetches a single entry from a collection by its slug.
+ *
+ * @param collection - The key of the collection to fetch the entry from
+ * @param id - The id of the entry to fetch
+ * @param locale - Optional SiteLocale to find the localized entry.
+ * @returns A promise that resolves to the requested collection entry or undefined if not found
+ */
+export async function getEntry<Key extends CollectionKey>(
+  collection: Key,
+  id: string,
+  locale: SiteLocale | null = getLocale(),
+) {
+  const entry = await getAstroCollectionEntry(collection, combine(id, locale));
+  
+  // If the entry is not found, try to fetch it without locale
+  return (locale)
+    ? entry || await getAstroCollectionEntry(collection, id)
+    : entry;
+}
+
+export function combine(id: string, locale?: string | null) {
```

Comment:
- `split` returns an object. Maybe use the same signature here?
- Should `locale` be a `SiteLocale | null`?